Docs: Add note about ulimit in bash_profile (#22562)
grafana/grafana: contribute/developer-guide.md

Another alternative is to limit the files being watched. The directories that are watched for changes are listed in the `.bra.toml` file in the root directory. 

> In case you are on Mac and you set your `ulimit` in `.bash_profile`, it will fail silently when changing the value later via command line so you have to rise it in the `.bash_profile` and then restart your terminal to pick the new value.

Reviewer: Won't this problem be the same on all Unices, not just Mac? This wording is a bit confusing to me anyhow; I think we might want to rewrite this section a bit instead, and clarify that you should commit your `ulimit` to ~/.bash_profile.

Author: Could not test it on linux right now so not sure if it behaves the same. If you can confirm it does the definitely we should make it more general.

Reviewer: The problem should be the same for all Unices (I guess if you have `ulimit` it's safe to conclude you're on Unix?). When you run `ulimit`, it will only be for the current shell, and in order to persist the new value, you have to stick it in some shell initialization file (for example ~/.bash_profile).

The reason it's a typical problem on Mac, I think, is that the default ulimit is low (256?). If you have a too low ulimit, the solution should be the same on all systems.

Reviewer: So, to conclude, I suggest we rewrite this section to state that you should commit the new `ulimit` command to your shell initialization file. I think actually ~/.bashrc is preferred to ~/.bash_profile, since the latter is supposed to be only executed by login shells, which can cause confusion, since if you start new shells they won't pick up changes to ~/.bash_profile before you log in again.

Author: Well there are definitely differences between mac and other linuxes, for example the default terminal.app at least does not run bashrc on new terminal windows but still runs bash_profile. So thats why I am not sure if we can just say this will work on both (also in new mac update the default shell is zsh I think).

Reviewer: The problem itself (committing `ulimit` for new shells) is general to Unix. When it comes to the method for committing your ulimit, it isn't entirely dependent on your flavour of Unix, but on the shell you use (and it also sounds as if it depends on the terminal).

I think we need to write something relatively general on how to update your shell initialization file, and decide on a sane default (~/.bash_profile or ~/.bashrc, this is completely independent of ulimit). When I read a Red Hat manpage on ulimit, they tell you to update ~/.bashrc.

> In case you are on Mac and you set your `ulimit` in `.bash_profile`, it will fail silently when changing the value later via command line so you have to rise it in the `.bash_profile` and then restart your terminal to pick the new value.

Reviewer 2: ```suggestion
If you are on a Mac and you set your `ulimit` in `.bash_profile`, then it will fail silently when changing the value later using the command line. You must add it in the `.bash_profile` and then restart your terminal to pick the new value.
```

Reviewer: See above discussion. In my opinion, we're better off rewriting the section than adding this.
